### ISSUE TITLE
General protection fault in rc4test_amd64_sse2()

### ISSUE THREAD

**Issue by aircrack-ng** [aircrack-ng/aircrack-ng#1742]
**Reported by misterx on 3 Mar 2018 21:51 UTC**

When compiling #3034 with clang on debian testing and running tests, ```test-aircrack-ng-0005.sh``` fails:

```
export CC=clang
export CXX=clang++
autoreconf -i
make
make check
```

When debugging
```
valgrind ./src/aircrack-ng--generic test/wep_64_ptw.cap
==10275== Memcheck, a memory error detector
==10275== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==10275== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==10275== Command: ./src/aircrack-ng--generic test/wep_64_ptw.cap
==10275== 
Opening test/wep_64_ptw.cap
Read 65282 packets.

   #  BSSID              ESSID                     Encryption

   1  00:12:BF:12:32:29  Appart                    WEP (30566 IVs)

Choosing first network as target.

Opening test/wep_64_ptw.cap
Attack will be restarted every 5000 captured ivs.
Starting PTW attack with 30566 ivs.
==10275== 
==10275== Process terminating with default action of signal 11 (SIGSEGV)
==10275==  General Protection Fault
==10275==    at 0x4194A9: rc4test_amd64_sse2 (in /root/aircrack-ng-trunk/src/aircrack-ng--generic)
==10275== 
==10275== HEAP SUMMARY:
==10275==     in use at exit: 9,131,626 bytes in 20,248 blocks
==10275==   total heap usage: 20,275 allocs, 27 frees, 11,733,390 bytes allocated
==10275== 
==10275== LEAK SUMMARY:
==10275==    definitely lost: 496 bytes in 1 blocks
==10275==    indirectly lost: 1,165,976 bytes in 20,235 blocks
==10275==      possibly lost: 288 bytes in 1 blocks
==10275==    still reachable: 7,964,866 bytes in 11 blocks
==10275==         suppressed: 0 bytes in 0 blocks
==10275== Rerun with --leak-check=full to see details of leaked memory
==10275== 
==10275== For counts of detected and suppressed errors, rerun with: -v
==10275== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
Segmentation fault
```

CPU is Intel Skylake. It happens with --avx, --avx2, --sse and --generic.

**Note**: it works just fine with gcc (tested gcc 7.3.0) on debian testing.

**Comment by aircrack-ng** [comment]
**Modified by misterx on 3 Mar 2018 21:51 UTC**

**Comment by aircrack-ng** [comment]
**Comment by misterx on 3 Mar 2018 21:51 UTC**

Fixed in #3052.
